tutorials/firstcodemod.md

```diff
@@ -155,6 +155,14 @@ namespace Celeste.Mod.Example {
 }
 ```
 
+### Hooking Methods
+There are currently 3 ways to hook methods. These are:
+
+* RuntimeDetour: This is currently used by [RainbowMod](https://github.com/EverestAPI/RainbowMod) and [MadelineEnergySelector](https://github.com/EverestAPI/MadelineEnergySelector), and will be deprecated for end-user mods once HookedMethod has a stable version.
+* HookedMethod.Hook: This is the replacement for RuntimeDetour, intended to be more user-friendly and have more features. Currently, it has nothing except a slightly simpler interface, and a backbone to easily allow adding IL-manipulation and complex non-detour functionality. RainbowMod will most likely be ported to HookedMethod.Hook in the near future. This is currently recommended for new mods that do not require a stable development interface. Documentation is available in [`Examples/Examples.cs`](https://github.com/EverestAPI/HookedMethod/blob/master/Examples/Program.cs) in the HookedMethod repo.
+* Everest.Events: This is the hooking method built in to Everest, and is used by [GhostMod](https://github.com/EverestAPI/GhostMod).
```

Comment:
`Everest.Events` aren't hooks at all. They're quite literally events that only get fired in specific occasions. Hooks allow you to completely replace the method or manipulate how it's called, which events don't allow.

Reply:
I feel like a hook is somewhat like a simplified breakpoint, running whenever what it hooks does. I feel like an event created by Everest counts as a hook, at least in how they're used. Neither definition of hook is exactly the dictionary definition.

Reply:
True, but we don't expose any method manipulation functionality that would be available with the other two methods. A newcomer could be confused why they can't replace the level loading method with `Everest.Events`, while being able to do so with the other two events.

Having a differentiation between `Everest.Events` and the other two methods is important IMO. If "events" isn't a good enough term, maybe "read-only hooks"?

Reply:
That could be good; although EventHooks, once fully complete, should have method manipulation functionality. It'll probably be limited to overriding arguments and return values, optionally bypassing the original function, though.

Reply:
I thought EventHooks would reflect the existing `Everest.Events`, allowing you to just += and -= delegates in a format very similar to the existing events? If someone wants to manipulate methods, they should just use HookedMethod IMO.

Reply:
They get passed a full HookedMethod instance, which will have those features for basic method manipulation inside other hooks. Detours will be optional once I find a good way to implement that.

Reply:
As long as the existing event listener method signatures (f.e. in GhostMod: `void OnLoadLevel(Level level, Player.IntroTypes playerIntro, bool isFromLoader)`) are natively supported, 👍